swing/src/main/java/at/bestsolution/fxembed/swing/FXEmbed.java

```diff
@@ -282,15 +290,22 @@ public static FXEmbed create(Consumer<Scene> consumer) throws Exception {
 				s.show();
 				long rawHandle = getWindowHandle(s);
 				embedder.setFXHandle(s, rawHandle, false);
-				consumer.accept(sc);
 				embedder.stage = s;
-				/*
-				 * the scene must be put into the stage only after reparenting,
-				 * otherwise popups will appear a few pixels too low (as if a window title bar was there) 
-				 */
-				s.setScene(sc);
+				
+				ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+				executor.schedule(() -> Platform.runLater(() -> {
+					/*
+					 * the scene must be put into the stage only after reparenting,
+					 * otherwise popups will appear a few pixels too low (as if a window title bar was there) 
+					 */
+					consumer.accept(sc);
+					s.setScene(sc);
+					embedder.resizeWindow(0);
+					executor.shutdown();
```

Comment:
according to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ExecutorService.html#shutdown() this could be done outside as tasks already scheduled will be executed.

Reply:
I thought that writing it this way would make the intention of having a "one-off" usage of the executor more explicit. But I can easily move it out... going to push this change in a moment.